### PR TITLE
Cleanup: Remove deprecated github.com/golang/protobuf usage

### PR DESCRIPTION
**Pull request by azylinski**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Replace deprecated github.com/golang/protobuf with google.golang.org/protobuf
- Regenerate cloudprovider/externalgrpc proto
- go mod tidy

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/externalgrpc/README.md [hunk 1]
```diff
@@ -56,8 +56,8 @@ To regenerate the gRPC code:
 1. install `protoc` and `protoc-gen-go-grpc`:
 
 ```bash
-go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.28
-go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.2
+go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.31.0
+go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.59.0
```
Comment (dbonfigli) [inline cluster-autoscaler/cloudprovider/externalgrpc/README.md:60]: are you sure this is correct? when i use this i get
```
$ go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.59.0
go: google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.59.0: module google.golang.org/grpc@v1.59.0 found, but does not contain package google.golang.org/grpc/cmd/protoc-gen-go-grpc
```
It seems to me that latest version is 1.3.0 and that seems what you used looking at https://github.com/kubernetes/autoscaler/blob/4b2dbe1ae956483f04af400671c7025030a530d5/cluster-autoscaler/cloudprovider/externalgrpc/protos/externalgrpc_grpc.pb.go#L18C1-L18C1

Comment (azylinski) [reply cluster-autoscaler/cloudprovider/externalgrpc/README.md]: Hi Diego, you're absolutely right - I've used package version, not the command itself.
`1.3.0` is the correct version: https://pkg.go.dev/google.golang.org/grpc/cmd/protoc-gen-go-grpc

**Comment by dbonfigli** [comment]
to make tests pass can you put back the previous license headers on pb.go files?

**Comment by azylinski** [comment]
> Member

I'd like to avoid manually editing the generated files.
I've added the `cloudprovider/externalgrpc/protos` to skip dirs on boilerplate verify.

**Review by dbonfigli — APPROVED** [review]

**Pull request merged**